lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -2483,17 +2507,20 @@ def _add_image_features(self, node_space: schema.NodeSpace) -> None:
                     image_info = self._get_image_info(
                         location, azure_runbook.marketplace
                     )
+                    generation = _get_vhd_generation(image_info)
+                    node_space.features.add(
```

Comment:
Make sure the test won't fail, if missing the two features.

Reply:
Private marketplace images will work like VHDs since there is no way to get info.

VHDs are already fine without defining architecture and hyper-v gen defaults to 1. If you want to use gen2 features on private marketplace image, you have to specify in the runbook that it is gen2 (similar to VHDs)